File: shoring/flywheel/src/test/java/com/panopset/tests/transformer/GlobalReplaceTransformerTest.java
package com.panopset.tests.transformer;

import java.io.IOException;

import org.junit.jupiter.api.Assertions;

import com.panopset.compat.Fileop;
import com.panopset.gp.GlobalReplaceProcessor;

public class GlobalReplaceTransformerTest extends SameFileTest {
	private final String searchStr;
	private final String replacementStr;
	private final String priorLineMustContain;

	public GlobalReplaceTransformerTest(String searchStr, String replacmentStr, String priorLineMustContain,
			String packageName, String refreshFile, String fromToFile, String expectedFile) {
		super(packageName, refreshFile, fromToFile, expectedFile);
		this.searchStr = searchStr;
		this.replacementStr = replacmentStr;
		this.priorLineMustContain = priorLineMustContain;
	}

	public GlobalReplaceTransformerTest(String searchStr, String replacmentStr, String packageName, String refreshFile,
			String fromToFile, String expectedFile) {
		this(searchStr, replacmentStr, "", packageName, refreshFile, fromToFile, expectedFile);
	}

	protected String xform() {
		try {
			getGlobalReplaceProcessor().process();
			return Fileop.readTextFile(tempFile);
		} catch (IOException e) {
			Assertions.fail("performTransformation failure.", e);
		}
		return "";
	}

	private GlobalReplaceProcessor grp;

	protected GlobalReplaceProcessor getGlobalReplaceProcessor() {
		if (grp == null) {
			grp = new GlobalReplaceProcessor(tempFile, searchStr, replacementStr, "txt", "", false);
			grp.setPriorLineMustContain(priorLineMustContain);
		}
		return grp;
	}
}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/transformer/GlobalReplaceTransformerTest.kt
package com.panopset.tests.transformer

import com.panopset.compat.Fileop
import com.panopset.gp.GlobalReplaceProcessor
import com.panopset.gp.PriorAndReplacementLineMustContainFilter
import org.junit.jupiter.api.Assertions
import java.io.IOException

class GlobalReplaceTransformerTest(
    private val searchStr: String, private val replacementStr: String, private val priorLineMustContain: String,
    packageName: String, refreshFile: String, fromToFile: String, expectedFile: String
) : SameFileTest(packageName, refreshFile, fromToFile, expectedFile) {

    override fun xform(): String {
        try {
            grp.process(PriorAndReplacementLineMustContainFilter
                (priorLineMustContain, "", searchStr, replacementStr))
            return Fileop.readTextFile(tempFile)
        } catch (e: IOException) {
            Assertions.fail<Any>("performTransformation failure.", e)
        }
        return ""
    }

    private var grp = GlobalReplaceProcessor(tempFile, "txt", false)
}
